packages/descendants/src/useInitDescendants.tsx

```diff
@@ -0,0 +1,32 @@
+import { Dispatch, useReducer } from 'react';
+
+import { DescendantsList } from './Descendants.types';
+import {
+  descendantsReducer,
+  DescendantsReducerAction,
+  DescendantsReducerType,
+  DescendantsState,
+} from './DescendantsReducer';
+
+export interface InitDescendantsReturnObject<T extends HTMLElement> {
+  descendants: DescendantsList<T>;
+  dispatch: Dispatch<DescendantsReducerAction<T>>;
+}
+
+/**
+ * Initializes a {@link DescendantsList}
+ */
+export const useInitDescendants = <
+  T extends HTMLElement,
+>(): InitDescendantsReturnObject<T> => {
+  const [state, dispatch] = useReducer<
+    React.Reducer<DescendantsState<T>, DescendantsReducerAction<T>>
```

Comment:
nit but since you're destructuring everything esle from React, maybe destructure Reducer as well